### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by JYamihud** [tkf/emacs-jedi#359]
What is the license? I can't find a single mention of a license. Please add a license file in the main directory. For now I gonna treat this software as proprietary. Until you add a license.

**Comment by immerrr** [comment]
> For now I gonna treat this software as proprietary.

I agree, that treating the software for which the license is unknown as proprietary/non-free is the most reasonable and safe approach.

However, one may find the license preambles in the source files to which they apply one by one:

- https://github.com/tkf/emacs-jedi/blob/e942a0e410cbb2a214c9cb30aaf0e47eb0895b78/jediepcserver.py#L6-L24
- https://github.com/immerrr/emacs-jedi/blob/e942a0e410cbb2a214c9cb30aaf0e47eb0895b78/jedi-core.el#L9-L21
- https://github.com/immerrr/emacs-jedi/blob/e942a0e410cbb2a214c9cb30aaf0e47eb0895b78/jedi.el#L9-L20

It may be tempting to _just_ add a license file to the main directory, licensing the entire project is not as simple. Whereas I may alter in any way I see fit _my_ contributions to the codebase and their respective license conditions, I cannot do the same for code that was added by other contributors. If the changes were made to previously non-explicitly-licensed files, I may need their explicit permission to _relicense_ their contributions from previously non-free/proprietary license to the license of my choosing. Tracking all contributors for a project that is 10 years old is tricky.

@syohex @tkf i'll defer to your judgement here, as you have been around for much longer than me.

**Comment by JYamihud** [comment]
>  I cannot do the same for code that was added by other contributors

GPL is good since if somebody contributes. Now the software is secured from YOU, the main developer, turning it into proprietary. Anyway, good to know that the file you showed is under GPL.

**Comment by asmeurer** [comment]
Well the very text you quoted says

> You should have received a copy of the GNU General Public License  along with Jedi EPC server. 

meaning the GPLv3 license should be actually included in the distribution (i.e., as a `LICENSE` or `COPYING` file in the repo). 

Also, I seriously doubt any contributor ever expected their contributions to be anything other than GPLv3.
